typing/env.ml

```diff
@@ -3290,7 +3327,14 @@ let lookup_global_name_module_no_locks
 let lookup_ident_module (type a) (load : a load) ~errors ~use ~loc s env =
   let path, locks, data =
     match find_name_module ~mark:use s env.modules with
-    | res -> res
+    | path, locks, data -> begin
+        match quotation_locks_offset locks with
+        | None | Some 0 -> path, locks, data
+        | Some n ->
+          may_lookup_error errors loc env
+            (Incompatible_stage (Lident s, loc, env.stage,
+                                 Location.none, env.stage - n))
+    end
```

Comment:
(also see my comment on `Local_env_lock`)

This is a common pattern and should be factored out into something like:
```
check_quotation_offset path locks ~loc ~lid:(Lident s) env.stage;
path, locks, data
```
where the first function call might raise `Incompatible_stage`.